lib/transaction.js

```diff
@@ -122,6 +125,33 @@ module.exports = function transaction() {
   });
 };
 
+transaction.start = function start(modelBase) {
+  if (!modelBase) {
+    return Promise.reject(new Error('objection.transaction.start: provide at least one Model class to bind to the transaction'))
+  }
+
+  if (!utils.isSubclassOf(modelBase, Model)) {
+    return Promise.reject(new Error('objection.transaction.start: all but the last argument should be Model subclasses'));
+  }
+
+  var knex = modelBase.knex();
+
+  return new Promise(function(resolve, reject) {
+    knex.transaction(function (trx) {
+        modelBase = modelBase.knex(trx);
+
+      resolve({
+        commit: function() {
+          return trx.commit();
+        },
+        rollback: function() {
+          return trx.rollback();
+        }
+      });
+    }).catch(reject);
```

Comment:
Passing the reject method for `cathch` here never does anything. To use the transaction the promise has to already be resolved (with the transaction object). The resolved promise cannot change its state. Therefore the `catch(reject)` does nothing.